gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -194,10 +295,30 @@ public void commit()
   public void write(D record)
       throws IOException {
     Preconditions.checkState(!closed, "Writer already closed");
-    valueWriter.write(record, rowBatch);
+    this.orcFileWriterLock.readLock().lock();
```

Comment:
Did you verify that writer can be called concurrently?

Reply:
Yes